File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/servers/list/listeners/SelectServerOnClickListener.java
package com.lasthopesoftware.bluewater.client.servers.list.listeners;

import android.view.View;

import com.lasthopesoftware.bluewater.client.browsing.library.access.session.IBrowserLibrarySelection;
import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library;
import com.lasthopesoftware.bluewater.client.connection.selected.InstantiateSelectedConnectionActivity;

/**
 * Created by david on 1/24/16.
 */
public class SelectServerOnClickListener implements View.OnClickListener {
	private final Library library;
	private final IBrowserLibrarySelection browserLibrarySelection;

	public SelectServerOnClickListener(Library library, IBrowserLibrarySelection browserLibrarySelection) {
		this.library = library;
		this.browserLibrarySelection = browserLibrarySelection;
	}

	@Override
	public void onClick(View v) {
		browserLibrarySelection.selectBrowserLibrary(library.getLibraryId());

		InstantiateSelectedConnectionActivity.startNewConnection(v.getContext());
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/servers/list/listeners/SelectServerOnClickListener.kt
package com.lasthopesoftware.bluewater.client.servers.list.listeners

import android.view.View
import com.lasthopesoftware.bluewater.client.browsing.library.access.session.SelectBrowserLibrary
import com.lasthopesoftware.bluewater.client.browsing.library.repository.Library
import com.lasthopesoftware.bluewater.client.connection.selected.InstantiateSelectedConnectionActivity.Companion.startNewConnection
import com.lasthopesoftware.bluewater.shared.promises.extensions.LoopedInPromise

class SelectServerOnClickListener(
	private val library: Library,
	private val browserLibrarySelection: SelectBrowserLibrary
) : View.OnClickListener {
	override fun onClick(v: View) {
		browserLibrarySelection.selectBrowserLibrary(library.libraryId)
			.eventually(LoopedInPromise.response({ startNewConnection(v.context) }, v.context))
	}
}
